tiledb/sm/filter/filter_create.cc

```diff
@@ -81,6 +82,15 @@ tiledb::sm::Filter* tiledb::sm::FilterCreate::make(FilterType type) {
       return tdb_new(tiledb::sm::FloatScalingFilter);
     case tiledb::sm::FilterType::FILTER_XOR:
       return tdb_new(tiledb::sm::XORFilter);
+#ifdef TILEDB_WEBP
```

Comment:
I think the addition of this macro makes the code look kind of complicated. Possibly adding a compile time constant defined webp_filter_exists or something like that and then only using the macro once to define that as true or false, and then you can do something like:

```
if constexpr (webp_filter_exists) {
// f it exists
} else {
// other thingo if it doesn't
}
```
You would define the constant itself like:
#ifdef TILEDB_WEBP
constexpr bool webp_filter_exists = true;
#else
constexpr bool webp_filter_exists = false;
#endif

(see platform.h maybe for a similar case).

Reply:
I would appreciate if you did this and then eliminated all uses of #ifdef TILEDB_WEBP in the C++ code.

Reply:
Thanks for the review! I agree this looks much better :+1: Still some work to do but should have updates pushed shortly